### PR TITLE
Flow - add pk/inserted & modified dates/_invocation id

### PR DESCRIPTION
**Pull request by WHYTEWYLL**

### CODE REVIEW

File: models/gold/core/core__dim_contract_labels.sql [hunk 1]
```diff
@@ -34,7 +40,21 @@ FINAL AS (
         streamline
 )
 SELECT
-    *
+    COALESCE (
```
Comment (forgxyz) [inline models/gold/core/core__dim_contract_labels.sql]: The order of columns determines how the data is returned to the user. While specific IDs (like block or tx or event) tend to be the first thing we list, for these PKs, I think it is more useful to the end user if we serve them at the end alongside the inserted and modified timestamp. These fields will not be used by 99% of studio users, just enterprise.

File: models/silver/nft/silver__nft_sales_s.sql [hunk 2]
```diff
@@ -105,6 +117,12 @@ combo AS (
         currency,
         tx_succeeded,
         _inserted_timestamp,
+        {{ dbt_utils.generate_surrogate_key(
```
Comment (forgxyz) [inline models/silver/nft/silver__nft_sales_s.sql:120]: The gold view is tested for uniqueness on the following columns, so tx id is not truly unique. We should use the following combination



```    
dbt_utils.unique_combination_of_columns:
          combination_of_columns:
            - tx_id
            - seller
            - buyer
            - nft_collection
            - nft_id
```

File: models/silver/nft/silver__nft_transactions_secondary_market_s.sql [hunk 3]
```diff
@@ -496,11 +496,23 @@ dedupe_gaia AS (
         ) = 1
 )
 SELECT
-    *
+    *,
+    {{ dbt_utils.generate_surrogate_key(
+        ['tx_id']
+    ) }} AS transactions_secundary_market_id,
```
Comment (forgxyz) [inline models/silver/nft/silver__nft_transactions_secondary_market_s.sql:502]: Secondary

File: models/silver/nft/silver__nft_transactions_secondary_market_s.sql [hunk 4]
```diff
@@ -496,11 +496,23 @@ dedupe_gaia AS (
...
+    '{{ invocation_id }}' AS _invocation_id
 FROM
     FINAL
 EXCEPT
 SELECT
-    *
+    *,
+    {{ dbt_utils.generate_surrogate_key(
+        ['tx_id']
+    ) }} AS transactions_secundary_market_id,
```
Comment (forgxyz) [inline models/silver/nft/silver__nft_transactions_secondary_market_s.sql:513]: Secondary

File: models/silver/nft/silver__nft_transactions_secondary_market_s.yml [hunk 5]
```diff
@@ -165,3 +165,15 @@ models:
 
       - name: _inserted_timestamp
         description: "{{ doc('_inserted_timestamp') }}"
+
+      - name: transactions_secundary_market_id
```
Comment (forgxyz) [inline models/silver/nft/silver__nft_transactions_secondary_market_s.yml:169]: Secondary

File: models/silver/staking/silver__staking_actions_s.sql [hunk 6]
```diff
@@ -106,7 +106,13 @@ FINAL AS (
         action,
         amount,
         node_id,
-        _inserted_timestamp
+        _inserted_timestamp,
+        {{ dbt_utils.generate_surrogate_key(
```
Comment (forgxyz) [inline models/silver/staking/silver__staking_actions_s.sql:110]: I think add event index and/or action

File: models/silver/transfers/silver__bridge_blocto_s.sql [hunk 7]
```diff
@@ -280,7 +280,13 @@ FINAL AS (
         END AS teleport_direction,
         l.blockchain,
         bridge,
-        _inserted_timestamp
+        _inserted_timestamp,
+        {{ dbt_utils.generate_surrogate_key(
+            ['tx_id']
+        ) }} AS bridge_blockto_id,
```
Comment (forgxyz) [inline models/silver/transfers/silver__bridge_blocto_s.sql:286]: Blocto

File: models/silver/transfers/silver__bridge_blocto_s.yml [hunk 8]
```diff
@@ -121,3 +121,15 @@ models:
 
       - name: _inserted_timestamp
         description: "{{ doc('_inserted_timestamp') }}"
+
+      - name: bridge_blockto_id
```
Comment (forgxyz) [inline models/silver/transfers/silver__bridge_blocto_s.yml:125]: Blocto

**Review by forgxyz — CHANGES_REQUESTED** [review]

File: models/gold/nft/nft__dim_allday_metadata.sql [hunk 9]
```diff
@@ -72,11 +77,43 @@ WITH allday AS (
...
+    play_type,
+    moment_date,
+    series,
+    set_name,
+    video_urls,
+    moment_stats_full,
+    COALESCE (
+        nft_unique_id,
+        {{ dbt_utils.generate_surrogate_key(['nft_id']) }}
+    ) AS dim_allday_metadata,
```
Comment (forgxyz) [inline models/gold/nft/nft__dim_allday_metadata.sql:109]: dim_allday_metadata_id

File: models/gold/nft/nft__dim_allday_metadata.yml [hunk 10]
```diff
@@ -160,3 +160,12 @@ models:
               column_type_list:
                 - OBJECT
                 - VARIANT
+
+      - name: DIM_ALLDAY_METADATA
```
Comment (forgxyz) [inline models/gold/nft/nft__dim_allday_metadata.yml:164]: _id

File: models/gold/nft/nft__dim_topshot_metadata.sql [hunk 11]
```diff
@@ -30,7 +30,10 @@ WITH topshot AS (
         video_urls,
         moment_stats_full,
         player_stats_game,
-        player_stats_season_to_date
+        player_stats_season_to_date,
+        nft_moment_metadata_topshot_id,
```
Comment (forgxyz) [inline models/gold/nft/nft__dim_topshot_metadata.sql:34]: dim_topshot_metadata_id

Comment (forgxyz) [reply models/gold/nft/nft__dim_topshot_metadata.sql:34]: + upd yml

File: models/gold/nft/nft__dim_ufc_strike_metadata.sql [hunk 12]
```diff
@@ -1,12 +1,6 @@
 {{ config(
     materialized = 'view',
-    meta={
-    'database_tags':{
-        'table': {
-            'PURPOSE': 'NFT, UFCSTRIKE'
-            }
-        }
-    },
+    meta ={ 'database_tags':{ 'table':{ 'PURPOSE': 'NFT, UFCSTRIKE' }}},
```
Comment (forgxyz) [inline models/gold/nft/nft__dim_ufc_strike_metadata.sql:3]: Even tho this is a CSV share, let's add the new PK column as a pipeline will be built out

Comment (forgxyz) [reply models/gold/nft/nft__dim_ufc_strike_metadata.sql:3]: + upd yml

**Review by forgxyz — CHANGES_REQUESTED** [review]
Silver all ran with no issue, 1 error on gold


```
16:30:50  Completed with 1 error and 0 warnings:
16:30:50  
16:30:50  Database Error in model nft__dim_topshot_metadata (models/gold/nft/nft__dim_topshot_metadata.sql)
16:30:50    000904 (42000): SQL compilation error: error line 25 at position 8
16:30:50    invalid identifier 'NFT_MOMENT_METADATA_TOPSHOT_ID'
16:30:50  
16:30:50  Done. PASS=16 WARN=0 ERROR=1 SKIP=0 TOTAL=17
```

**Comment by WHYTEWYLL** [comment]
@forgxyz did you ran `dbt run -m models/silver/nft/metadata/livequery/silver__nft_topshot_metadata.sql` before running the gold view?

File: models/gold/nft/nft__dim_ufc_strike_metadata.sql (hunk 12 above)
Comment (forgxyz) [reply models/gold/nft/nft__dim_ufc_strike_metadata.sql:3]: Yes, but we are adding a PK to every model with the naming convention `<view>_id` so we should do that consistently across the board

**Review by forgxyz — APPROVED** [review]

**Pull request merged**